plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/hooks/use-product-type-selector/index.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+/**
+ * Internal dependencies
+ */
+import { store as woocommerceTemplateStateStore } from '../../store';
+import type { ProductTypeProps } from '../../types';
+
+type ProductTypeSelector = {
+	productTypes: ProductTypeProps[];
+	current: ProductTypeProps;
+	set: ( productType: string ) => void;
+};
+
+/**
+ * Hook to get data from the store to manage the product type selector.
+ * Also, it provides a function to switch the current product type.
+ * It's a layer of abstraction to the store.
+ *
+ * @return {ProductTypeSelector} The product type selector data and functions.
+ */
+export default function useProductTypeSelector(): ProductTypeSelector {
+	/*
+	 * Get the product types and the current product type
```

Comment:
Nitpick: Can we modify this comment to be more specific? Maybe: `Get the available product types and the current product type`.